site/content/blog/terraform-carvel-provider.md

```diff
@@ -0,0 +1,151 @@
+---
+title: "Using Carvel Terraform Provider to manage Kubernetes workloads"
+slug: terraform-carvel-provider
+date: 2021-11-17
+author: Soumik Majumder
+excerpt: "Looking to leverage Carvel tools to manage workloads on Kubernetes while setting up your platform using Terraform? Carvel's Terraform provider has your back"
+image: /img/logo.svg
+tags: ['Soumik Majumder']
+---
+
+The Carvel tools are designed to empower our users to manage their Kubernetes workloads effectively. We realise that engineers leveraging Terraform to declaratively define their platforms might want to use Carvel tools to set up applications and workloads on their Kubernetes clusters in a predictable manner. Carvel's [terraform provider](https://github.com/vmware-tanzu/terraform-provider-carvel) allows engineers to do _exactly_ this using Terraform configurations.
+
+In this blog, we will be using the provider to deploy this [sample guestbook application](https://github.com/vmware-tanzu/terraform-provider-carvel/tree/develop/examples/guestbook/ytt-config) on a kubernetes cluster.
+
+Do make a copy of the `/guestbook/ytt-config` folder in your working directory if you wanna follow along!
+
+## Setting things up
+The [Carvel provider](https://registry.terraform.io/providers/vmware-tanzu/carvel/latest) is published on the Terraform registry.
+
+The resources used require their respective binaries to be available on `$PATH`. We will be using _ytt_ to apply an overlay to our manifests and then deploy the resources using _kapp_. We can make the Carvel binaries available by using our install script
+
+```bash
+$ wget -O- https://carvel.dev/install.sh | bash
+# or with curl...
+$ curl -L https://carvel.dev/install.sh | bash
+```
+
+See our [**Install** section](/) for alternative installation methods.
+
+We will be using the latest version of the Carvel provider available as of today.
+
+```terraform
+terraform {
+  required_providers {
+    carvel = {
+      source = "vmware-tanzu/carvel"
+      version = "0.10.0"
+    }
+  }
+}
+```
+
+## Connecting to the cluster
+The _kapp_ resource requires the `kubeconfig` attribute to connect to the cluster. We will be asking the _kapp_ resource to use the config used by _kubectl_ in the same environment. 
+
+Look [here](https://github.com/vmware-tanzu/terraform-provider-carvel/blob/develop/docs/provider.md) for more ways of authenticating while connecting to a cluster.
+
+```terraform
+provider "carvel" {
+  kapp {
+    kubeconfig {
+      from_env = true
+    }
+  }
+}
+```
+## Overlaying using _ytt_
+We can use [_ytt_](https://github.com/vmware-tanzu/carvel-ytt) to template and overlay our manifests. In this blog, we will be asking _ytt_ to apply an overlay that ensures that all Deployment resources spawn 1 replica.
+
+```terraform
+data "carvel_ytt" "guestbook" {
+  files = ["ytt-config"]
+  ignore_unknown_comments = true
```

Comment:
lets not have `ignore_unknown_comments` in here. it shouldnt be used in 99% cases

Reply:
Made this change!